lib/psych/visitors/yaml_tree.rb

```diff
@@ -310,19 +310,20 @@ def visit_String o
           style = Nodes::Scalar::LITERAL
           plain = false
           quote = false
-        elsif o =~ /\n/
-          style = Nodes::Scalar::LITERAL
         elsif o == '<<'
           style = Nodes::Scalar::SINGLE_QUOTED
           tag   = 'tag:yaml.org,2002:str'
           plain = false
           quote = false
+        elsif o =~ /\n[^\Z]/  # match \n except blank line at the end of string
+          style = Nodes::Scalar::LITERAL
+        elsif o.length > line_width
```

Comment:
> Can you do the same benchmark with benchmark-ips?

```
Calculating -------------------------------------
      variable:false    78.894k i/100ms
        lambda:false    84.828k i/100ms
       variable:true    85.861k i/100ms
         lambda:true    76.106k i/100ms
-------------------------------------------------
      variable:false      3.947M (± 6.7%) i/s -     19.724M
        lambda:false      3.987M (± 6.3%) i/s -     19.850M
       variable:true      4.568M (± 5.3%) i/s -     22.839M
         lambda:true      2.828M (± 6.1%) i/s -     14.156M
```

The conclusion is the same. When `line_width` is not set, then difference between variable and lambda is NOT statistically significant. When the `line_width` is set, then variable is significantly faster.

> Also note that we could refactor to a simple method call rather than a lambda:

I hope that you’re not serious…